src/main/scala/com/campudus/tableaux/router/MediaRouter.scala

```diff
@@ -0,0 +1,116 @@
+package com.campudus.tableaux.router
+
+import java.util.UUID
+
+import com.campudus.tableaux.TableauxConfig
+import com.campudus.tableaux.controller.MediaController
+import com.campudus.tableaux.database.domain.DomainObject
+import com.campudus.tableaux.database.model.FolderModel.FolderId
+import org.vertx.scala.core.http.HttpServerRequest
+import org.vertx.scala.core.json.JsonObject
+import org.vertx.scala.router.RouterException
+import org.vertx.scala.router.routing._
+
+import scala.concurrent.Promise
+import scala.util.matching.Regex
+
+sealed trait FileAction
+
+case class UploadAction(fileName: String,
+                        mimeType: String,
+                        exceptionHandler: (Throwable => Unit) => _,
+                        endHandler: (() => Unit) => _,
+                        streamToFile: (String) => _) extends FileAction
+
+object MediaRouter {
+  def apply(config: TableauxConfig, controllerCurry: (TableauxConfig) => MediaController): MediaRouter = {
+    new MediaRouter(config, controllerCurry(config))
+  }
+}
+
+class MediaRouter(override val config: TableauxConfig, val controller: MediaController) extends BaseRouter {
+  /**
+   * Regex for a UUID Version 4
+   */
+  val uuidRegex: String = "[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}"
+
+  val FolderId: Regex = "/folders/(\\d+)".r
+  val FileId: Regex = s"/files/($uuidRegex)".r
+  val FileIdStatic: Regex = s"/files/($uuidRegex)/.*".r
+
+  override def routes(implicit req: HttpServerRequest): Routing = {
```

Comment:
I think it would make sense to put the requests into their own methods, so this method here contains more or less only:
```
case Get("/something") => asyncReply(whatever(req))
case Post("/something") => asyncReply(somethingElse(req))
...
```

Especially when looking at the file upload, it's just a lot going on in this method here.